### ISSUE TITLE
Python 3.8.2 doesn't work with mypy linting

### ISSUE THREAD

**Issue by mavysavydav** [feast-dev/feast#1631]
## Expected Behavior 
Normal linting checks.
## Current Behavior
Errors on type comments and doesn't proceed with rest of lint checking.
`(.venv2) [tw-mbp-davidl feast (master)]$ make lint-python
cd /Users/davidl/workspace/feast/sdk/python; mypy feast/ tests/

feast/client.py:411: error: invalid syntax

Found 1 error in 1 file (checked 109 source files)

make: *** [lint-python] Error 2`

## Steps to reproduce
Use Python 3.8.2 and run make lint-python.
### Specifications

- Version: 0.10.7
- Platform: Mac
- Subsystem: N/A

## Possible Solution
Explore alternatives to type comments. Or possibly there's a setting in mypy (or version of mypy) that can work with 3.8.2 better.  Maybe type comments are not encouraged in this Python version. Have not investigated.

**Comment by woop** [comment]
Which version of Mypy are you using?

**Comment by codyjlin** [comment]
> Which version of Mypy are you using?

@woop 

```shell
❯ mypy --version
mypy 0.790
❯ python --version
Python 3.8.2
```

**Comment by codyjlin** [comment]
I bypassed this by using `python 3.7.4` instead. (I followed [this article](https://www.freecodecamp.org/news/manage-multiple-python-versions-and-virtual-environments-venv-pyenv-pyvenv-a29fb00c296f/) on how to manage different python versions with pyenv).

It could be worth documenting for now that one should use a particular python version (instead of "3.7 and above") for now in the [`CONTRIBUTING.md` docs](https://github.com/feast-dev/feast/blob/master/CONTRIBUTING.md#environment-setup).  A more expensive fix would be to update all the type annotations to move away from type comments (it seems like type comments are the old way of doing things from [the docs](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html))
